app/controllers/restaurants_controller.rb

```diff
@@ -0,0 +1,41 @@
+class RestaurantsController < ApplicationController
+  def create
+    restaurant = Restaurant.new(restaurant_params)
+    if !restaurant.valid?
```

Comment:
whats better to do do is 
```
unless restaurant.save!
    respond_to do |format|
        format.json { render json: restaurant.errors.messages, status: 400 }
    end
end
```
This way you remove the negative condition (negative conditions are frowned upon in the rails community - check out the our the [Air BNB style guide](https://github.com/airbnb/ruby). And [our styleguide](https://github.com/WeConnect/style-guide/blob/0da00445f23d0ca627c3516c5941d234d8b91474/ruby/README.md)